### DISCUSSION TITLE
global SkipMemoryCache

### DISCUSSION THREAD

**Discussion by moadib2k** [ZiggyCreatures/FusionCache#176] [Q&A]
My backend is running with two different types of clients. One part is running in long lasting containers, the other part is running in lambdas.  I saw the addition of SkipMemoryCache in .22 and put a test implementation in place. 

I do not think this feature does what I was thinking.  I am attempting to have my containers use the memory cache backed by the distributed cache and to have my lambdas use skip the memory cache and only use the distributed cache.  These are two different parts of the system sharing the same cache data.  

This feature appears to work on a cache item entry level.

This is my config

> new FusionCacheEntryOptions
> {
>       Duration = TimeSpan.FromMinutes(5),
>       IsFailSafeEnabled = true,
>       FailSafeMaxDuration = TimeSpan.FromMinutes(10),
>       FailSafeThrottleDuration = TimeSpan.FromSeconds(30),
>       FactorySoftTimeout = TimeSpan.FromMilliseconds(100),
>       FactoryHardTimeout = TimeSpan.FromSeconds(30),
>       DistributedCacheSoftTimeout = TimeSpan.FromSeconds(1),
>       DistributedCacheHardTimeout = TimeSpan.FromSeconds(2),
>       AllowBackgroundDistributedCacheOperations = true,
>       JitterMaxDuration = TimeSpan.FromSeconds(2),
>       SkipMemoryCache = true
> }

Initially, I set up fusion with no memory cache (did not register the type) and received an error.  I added the memory cache registration back in and looking at the logs on item creation, it appears that it still adds the entry to the memory cache.
> [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLS K=TestKey): calling Set<T> ZiggyCreatures.Caching.Fusion.FusionCache - > > [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLS K=TestKey): saving entry in memory 
> [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLS K=TestKey): serializing the entry
> [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLS K=TestKey): setting the entry in distributed 

When retrieving the item it looks like it only checks the distributed cache
> [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RM0 K=TestKey): calling TryGet<T> null
> [VRB] FUSION [N=FusionCache] (O=0HMU0VQDK7RM0 K=TestKey): trying to get entry from distributed
> [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RM0 K=TestKey): distributed entry found 
> [VRB] FUSION [N=FusionCache] (O=0HMU0VQDK7RM0 K=TestKey): using distributed entry
> [DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RM0 K=TestKey): return SUCCESS

Also, when I get an entry that was added to distributed cache from a container that had items set to store in both it gets them from memory first. 

[DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLU K=/TradeCache/MYSQL): calling GetOrSet<T> FEO[LKTO=/ DUR=04:00:00 SKM=N SKD=N SKDRWS=N DDUR=/ JIT=0 PR=N FS=Y FSMAX=08:00:00 DFSMAX=/ FSTHR=30s FSTO=50ms FHTO=/ TOFC=Y DSTO=/ DHTO=/ ABDO=N SBN=N BBO=Y]
[VRB] FUSION [N=FusionCache] (O=0HMU0VQDK7RLU K=/Key): trying to get from memory
[DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLU K=/Key): memory entry found 
[VRB] FUSION [N=FusionCache] (O=0HMU0VQDK7RLU K=/Key): using memory entry
[DBG] FUSION [N=FusionCache] (O=0HMU0VQDK7RLU K=/Key): return

**Comment by jodydonetti** [comment]
Hi @moadib2k and thanks for using FusionCache!

What you are describing seems in fact strange, and not the expected behavior. Having said that I have tests that specifically verify the `SkipMemoryCache` option, and they seem fine.

Would you be able to provide me with a [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example)? That would really help me in troubleshooting your scenario.

Thanks.

**Reply by jodydonetti** [reply]
Hi @moadib2k , any news on this?
